#18356 Added description of requires boolean in package-lock.json to …

kreig303 opened the pull request:
…npm docs.

Referenced in #18356.

zkat committed:
npm-lifecycle@1.0.3

zkat committed:
npm-packlist@1.1.9

kreig303 committed:
#18356 Added description of requires boolean in package-lock.json to npm docs.

iarna committed:
unsupported: Fix formatting of unsupported version warning

iarna committed:
npm-profile@2.0.4

iarna committed:
qw@1.0.1

iarna committed:
cli-table2@0.2.0

iarna committed:
is-cidr@1.0.0

iarna committed:
qrcode-terminal@0.11.0

iarna committed:
query-string@5.0.0

iarna committed:
npm-registry-client@8.5.0

iarna committed:
install.sh: Drop upgrading from npm@1 support

This is fine as npm@5 doesn't run on versions of Node.js that ship
npm@1.

Credit: @iarna

kevinburke commented on doc/files/package-lock.json.md:
```diff
@@ -46,6 +46,9 @@ The version of the package this is a package-lock for. This must match what's in
 An integer version, starting at `1` with the version number of this document
 whose semantics were used when generating this `package-lock.json`.
 
+### requires
```
I'm not sure that this is accurate, here's what the package-lock.json file looks like for me (generated using 5.4.2):

```patch
     "@types/supertest": {
       "version": "2.0.3",
       "resolved": "https://registry.npmjs.org/@types/supertest/-/supertest-2.0.3.tgz",
       "integrity": "sha512-QxgjuDhQEq4tPJXTOnGoN4P/BWwvftvkaCZhXMp1C1Otk9kGXb7l/wA7kwz3MwtEMnovdYikS+ZtvaQ4RbhF4g==",
-      "dev": true
+      "dev": true,
+      "requires": {
+        "@types/superagent": "3.5.5"
+      }
     },
```

In other words, it's an object, not a boolean.

kevinburke replied:
My apologies, it looks like it's specified in the file in two different places; at the top level, it's a boolean, inside of a package descriptor, it's an object. Not sure the best way to handle that.

kreig303 replied:
So how should we handle this?  Some docs are better than none... and if this isn't to be accepted I'll just close it.

iarna committed:
profile,token: Add new profile commands

iarna committed:
auth/legacy: Use new profile lib for login

iarna committed:
map-to-registry: Generally add support for otp flag

iarna committed:
error-message: Add custom error messages

iarna committed:
publish: Prompt for OTP when required and we have a TTY

iarna committed:
request@2.83.0

A slew of dependency updates, include an update to tough-cookie with a
security fix.

iarna committed:
abbrev@1.1.1

License change.

iarna committed:
aproba@1.2.0

Browser support

iarna committed:
meant@1.0.1

Doc updates.

iarna committed:
rimraf@2.6.2

Fixes a long standing bug in rimraf's attempts to work around Windows limitations
where it owns a file and can change its perms but can't remove it without
first changing its perms.

iarna committed:
doc: changelog for 5.5.0

iarna committed:
5.5.0

iarna committed:
auth: Fix OTP detection

iarna committed:
doc: changelog for 5.5.1

iarna committed:
5.5.1

kreig303 commented:
Closing.  Out of sync and need to properly PR.
